counterparty-core/counterpartycore/lib/messages/detach.py

```diff
@@ -0,0 +1,143 @@
+import logging
+import struct
+
+from counterpartycore.lib import config, exceptions, ledger, script, util
+
+logger = logging.getLogger(config.LOGGER_NAME)
+
+ID = 102
+
+
+def validate(source, destination=None):
+    problems = []
+
+    # check if source is a UTXO
+    if not util.is_utxo_format(source):
+        problems.append("source must be a UTXO")
+
+    # check if destination is an address
+    if destination is not None:
+        try:
+            script.validate(destination)
+        except script.AddressError:
+            problems.append("destination must be an address")
```

Comment:
If the destination is an invalid address, either burning all the assets or returning them to the address associated with the UTXO to avoid assets remaining in a spent UTXO

Reply:
done here https://github.com/CounterpartyXCP/counterparty-core/pull/2626/commits/82ecd19af35ffabf43577567b08b7477d6b7f561